lib/serial_terminal.pm

```diff
@@ -122,8 +122,15 @@ wait_serial(get_login_message(), 300);
 
 sub get_login_message {
     my $arch = get_required_var("ARCH");
-    return is_sle() ? qr/Welcome to SUSE Linux Enterprise .*\($arch\)/
-      : is_sle_micro() ? qr/Welcome to SUSE Linux.* Micro .*\($arch\)/
+    my $agama_opts = get_var('AGAMA_PROFILE_OPTIONS', '');
+
+    if (is_sle()) {
+        if ($agama_opts =~ /software_only_required=true/) {
+            return qr/\blogin:/;
+        }
+        return qr/Welcome to SUSE Linux Enterprise .*\($arch\)/;
+    }
+    return is_sle_micro() ? qr/Welcome to SUSE Linux.* Micro .*\($arch\)/
```

Comment:
the code might be equivalent replacing ternary `else` with if + returns, but could you please keep using ternaries so you don't need to change existing lines, you can have one:
 `is_sle() && $agama_opts =~ /software_only_required/`